### PR TITLE
chore(core): Upgrade configcat-js and configcat-node

### PR DESCRIPTION
**Pull request by eirikurn**
## What

Upgrade configcat packages.
Make the feature flag types more generic to match the upgraded configcat types.

## Why

To stay up to date and lower our requests per month.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/react/feature-flags/src/lib/useFeatureFlag.ts [hunk 1]
```diff
@@ -28,7 +32,7 @@ export const useFeatureFlag = <T extends boolean | string>(
       .getValue(featureFlag, defaultValue, user)
       .then((value) => {
         if (mounted) {
-          setState({ value: value as T, loading: false })
+          setState({ value: value as SettingTypeOf<T>, loading: false })
```
Comment (snaerth) [inline libs/react/feature-flags/src/lib/useFeatureFlag.ts:35]: Does TS not infer the value?

File: libs/feature-flags/src/lib/configcat.ts [hunk 2]
```diff
@@ -15,18 +22,20 @@ export class Client implements FeatureFlagClient {
...
-      this.configcat = eval('require')('configcat-node').createClient(
+      this.configcat = eval('require')('configcat-node').getClient(
         resolvedSdkKey,
+        null,
         ccConfig,
       )
     } else {
       // eslint-disable-next-line @typescript-eslint/no-var-requires
-      this.configcat = require('configcat-js').createClient(
+      this.configcat = require('configcat-js').getClient(
```
Comment (snaerth) [inline libs/feature-flags/src/lib/configcat.ts:36]: Is the updated package, not an ES module? Just wondering if import() should be used instead

**Review by snaerth — APPROVED** [review]
Awesome work! 💯 
How does this save us requests per month?

File: libs/feature-flags/src/lib/configcat.ts (hunk 2 above)
Comment (eirikurn) [reply libs/feature-flags/src/lib/configcat.ts:36]: This module is dynamically importing web vs nodejs version of configcat. We're trying not to run or bundle both.

**Review by snaerseljan — APPROVED** [review]

**Review by snaerseljan — APPROVED** [review]

**Pull request merged**